docs/docs/en/user-guide/adapters/fastapi/index.md

```diff
@@ -1,3 +1,153 @@
 # FastAPI
 
-**FastAPI** allows for a fairly scalable setup using [**FastAPI**](https://fastapi.tiangolo.com/){target="_blank"} to execute your workflows. It supports execution with multiple workers, with each workflow being executed in the context of a WebSocket connection.
+The **FastAPI Adapter** allows you to expose your FastAgency workflows as a REST API using the [**FastAPI**](https://fastapi.tiangolo.com/){target="_blank"} framework.
+
+## Use Cases
+
+This section outlines the scenarios where it's particularly beneficial to use the **FastAPI** Adapter.
+
+### When to Use the FastAPI Adapter
+
+- **Custom Client Applications**: If you want to build your **own client applications** that interact with your FastAgency workflows using REST APIs, this adapter provides greater flexibility and control over the client-side implementation.
+- **Moderate User Demand**: The FastAPI Adapter is a good fit for scenarios with **moderate user request volume**. For example, it's well-suited for a medium-sized company developing an internal custom chat application.
+- **Simplified Production Setup**: Choose this adapter if you need a **simple and easy-to-manage** production setup for deploying your FastAgency workflows as a REST API.
+
+
+## Architecture Overview
+
+The following diagram illustrates the high-level architecture of an application using the  **FastAPI Adapter with Mesop UI** as its frontend:
+
+![Mesop FastAPI](../images/mesop_fastapi.png)
```

Comment:
maybe in the diagram, we could beside Mesop draw an "Custom application", so that we show that Mesop client is not mandatory for this setup

Reply:
I have updated the images and the content to reflect the same